docs/reference/elements/py-config.md

```diff
@@ -0,0 +1,242 @@
+
+The `<py-config>` element should be placed within the `<body>` element.
+
+## Attributes
+
+| attribute | type | default | description |
+|----|----|----|----|
+| **type** | string | "toml" | Syntax type of the `<py-config>`. Value can be `json` or `toml`. Default: "toml" if type is unspecifed. |
+| **src** | url |    | Source url to an external configuration file. |
+
+## Examples
+
+- `<py-config>` using TOML (default)
+```html
+<py-config>
+  autoclose_loader = true
+
+  [[runtimes]]
+  src = "https://cdn.jsdelivr.net/pyodide/v0.21.2/full/pyodide.js"
+  name = "pyodide-0.21.2"
+  lang = "python"
+</py-config>
+```
+Note: `[[runtimes]]` is a TOML table. Make sure this is the last item within a py-config, as the properties created after it go into the runtimes object.
+
+- JSON config using the `type` attribute.
+```html
+<py-config type="json">
+  {
+    "autoclose_loader": true,
+    "runtimes": [{
+      "src": "https://cdn.jsdelivr.net/pyodide/v0.21.2/full/pyodide.js",
+      "name": "pyodide-0.21.2",
+      "lang": "python"
+    }]
+  }
+</py-config>
+```
+
+- Use of the `src` attribute:
+```
+<py-config src="./custom.toml"></py-config>
+```
+where `custom.toml` contains
+
+```
+autoclose_loader = true
+[[runtimes]]
+src = "https://cdn.jsdelivr.net/pyodide/v0.21.2/full/pyodide.js"
+name = "pyodide-0.21.2"
+lang = "python"
+```
+
+- JSON using the `type` attribute.
+```
+<py-config type="json" src="./custom.json"></py-config>
+```
+where `custom.json` contains
+
+```
+{
+  "autoclose_loader": true,
+  "runtimes": [{
+    "src": "https://cdn.jsdelivr.net/pyodide/v0.21.2/full/pyodide.js",
+    "name": "pyodide-0.21.2",
+    "lang": "python"
+  }]
+}
+```
+
+One can also use both i.e pass the config from `src` attribute as well as specify it as `inline`. So the following snippet is also valid:
+
+```
+<py-config src="./custom.toml">
+  paths = ["./utils.py"]
+</py-config>
+```
+
+This can also be done via JSON using the `type` attribute.
+
+```
+<py-config type="json" src="./custom.json">
+  {
+    "paths": ["./utils.py"]
+  }
+</py-config>
+```
+
+Note: While the `<py-config>` tag supports both TOML and JSON, one cannot mix the type of config passed from 2 different sources i.e. the case when inline config is in TOML format while config from src is in JSON format is NOT allowed. Similarly for the opposite case.
+
+---
+
+This is helpful in cases where a number of applications share a common configuration (which can be supplied via `src`), but their specific keys need to be customised and overridden.
+
+The keys supplied through `inline` override the values present in config supplied via `src`.
+
+One can also declare dependencies so as to get access to many 3rd party OSS packages that are supported by PyScript.
+You can also link to `.whl` files directly on disk like in our [toga example](https://github.com/pyscript/pyscript/blob/main/examples/toga/freedom.html).
```

Comment:
I'd add something like:

Package dependencies in the config can be declared by using the direct link to the package URL (whl or any other format supported by the chosen runtime) or by just providing the package name [and version]. If only the name [and version] are provided, packages will be installed directly from what's provided by your runtime or from PyPI. NOTICE that only pure python packages from PyPI will work and packages with C dependencies will not. These need to be built specifically for WASM (please, consult the Pyodide project for more information about what's supported and on how to build packages with c dependencies)

Reply:
added